### PR TITLE
Clarify recommendation on datagram acknowledgements

### LINKED ISSUE #47: Question about: "not used for loss recovery"
Can someone explain why the ack from Datagram-only packets are not used for loss recovery?

> Receivers SHOULD support delaying ACK frames (within the limits specified by max_ack_delay) in reponse to receiving packets that only contain DATAGRAM frames, since the timing of these acknowledgements is not used for loss recovery.

I would appreciate if someone can point me to an existing discussion or explanation. Thanks.

### PR DESCRIPTION
**Pull request by DavidSchinazi**
This was a suggestion from @MikeBishop.

Fixes #47

### CODE REVIEW

**Review by LPardue — APPROVED** [review]
This is an editorial change that clarifies the intent of the text.

**Review by ekinnear — APPROVED** [review]
Seems strictly more clear, nice!

File: draft-ietf-quic-datagram.md [hunk 1]
```diff
@@ -242,9 +242,12 @@ priorities to DATAGRAM frames with respect to each other and to QUIC streams.
 
 Although DATAGRAM frames are not retransmitted upon loss detection, they are
 ack-eliciting ({{!RFC9002}}). Receivers SHOULD support delaying
-ACK frames (within the limits specified by max_ack_delay) in reponse to receiving
-packets that only contain DATAGRAM frames, since the timing of these
-acknowledgements is not used for loss recovery.
+ACK frames (within the limits specified by max_ack_delay) in response to
+receiving packets that only contain DATAGRAM frames, since the sender takes no
+action if these packets are temporarily unacknowledged. Receivers SHOULD
```
Comment (tfpauly) [inline draft-ietf-quic-datagram.md]: We're not adding new normative behavior for acknowledgements, so I think we should just state the fact of how QUIC will behave here.

```suggestion
action if these packets are temporarily unacknowledged. Receivers will
```

File: draft-ietf-quic-datagram.md [hunk 2]
```diff
@@ -242,9 +242,12 @@ priorities to DATAGRAM frames with respect to each other and to QUIC streams.
...
+continue to send ACK frames when conditions indicate a packet might be lost,
+since the packet's payload is unknown to the receiver, and when dictated by
```
Comment (tfpauly) [inline draft-ietf-quic-datagram.md:249]: What does the "payload is unknown" mean?

**Review by tfpauly — COMMENTED** [review]
A few nits. I'm also wondering if we should just leave the old sentence ending with "is not used for loss recovery", and then explain it.

File: draft-ietf-quic-datagram.md (hunk 2 above)
Comment (DavidSchinazi) [reply draft-ietf-quic-datagram.md:249]: Unknown as in "you do not know the contents of a packet that you never received", so if you see packet numbers 1,2,3,5,6,7,8 you don't know if packet 4 was datagram-only or not, so you shouldn't treat 4 as datagram-only

Comment (tfpauly) [reply draft-ietf-quic-datagram.md:249]: Ah, got it.

**Review by tfpauly — APPROVED** [review]

**Pull request merged**